src/accountant_stub.rs

```diff
@@ -51,35 +52,38 @@ impl AccountantStub {
     /// Request the balance of the user holding `pubkey`. This method blocks
     /// until the server sends a response. If the response packet is dropped
     /// by the network, this method will hang indefinitely.
-    pub fn get_balance(&self, pubkey: &PublicKey) -> io::Result<Option<i64>> {
+    pub fn get_balance(&self, pubkey: &PublicKey) -> FutureResult<i64, i64> {
```

Comment:
How about something like`FutureResult<Option<i64>, io::Error>` ? Or perhaps the library offers a `futures::io::FutureResult<Option<i64>>`?